Userland/Libraries/LibWeb/CSS/Properties.json

```diff
@@ -1203,6 +1203,18 @@
       "unitless-length"
     ]
   },
+  "max-inline-size": {
+    "inherited": false,
+    "initial": "auto",
+    "valid-types": [
+      "length [0,∞]",
+      "percentage [0,∞]"
+    ],
+    "valid-identifiers": [
+      "auto",
+      "none"
+    ]
+  },
```

Comment:
This feels like something that should be written more like:

```
"max-inline-size": {
    "logical-alias-for": [
		"width",
		"height"
	],
},
```

Duplicating all the metadata for every logical property is going to get very noisy (and makes it easier to make mistakes).

Since we have a *lot* of these properties to deal with, it would be nicer to generate code for resolving them instead of doing it manually.

Reply:
> Duplicating all the metadata for every logical property is going to get very noisy..

This makes sense. I could add logic to reuse the metadata of the alias target.

>  it would be nicer to generate code for resolving them instead of doing it manually.

Do you mean to have in the Properties.json something like:

```json
"max-inline-size": {
  "logical-alias-for": [
    "max-width": {
      "conditions": [ { "writing-mode": [ "horizontal-tb" ] } ]
    },
    "max-height": {
      "conditions": [ true ]
    }
  ]
}
```

And generate code from that? Seems like it could unwieldy pretty quickly based on the complexity of the conditions.

Reply:
I don't think we need to go that far, but it would be nice to at least avoid repeating the set of valid values for each property :)

Reply:
I realized that I had got the `initial` value wrong for max-inline-size. So I also added the ability for the logical property to override something

```json
  "max-inline-size": {
    "logical-alias-for": "max-width",
    "__comment": "Also a logical alias for max-height",
    "initial": "none"
  }
```